lib/internal/bootstrap/node.js

```diff
@@ -175,6 +175,9 @@ if (credentials.implementsPosixCredentials) {
   process.getgid = credentials.getgid;
   process.getegid = credentials.getegid;
   process.getgroups = credentials.getgroups;
+  if (!credentials.isApple) {
```

Comment:
I'm not a fan of apis that may or may not exist depending on what platform you're on. My preference would be for the method to always be defined but throw a reasonable exception on unsupported platforms.

Reply:
I can't think of a similar case but probably there are some cases when it comes to compatibility with Windows? Maybe this could be handled like those cases then?
In general, I also like the idea of a consistent API with reasonable exceptions. :+1: